core/src/main/scala/org/http4s/Credentials.scala

```diff
@@ -65,9 +69,19 @@ final case class BasicCredentials(username: String, password: String) {
 }
 
 object BasicCredentials {
+  private val utf8Charset = JavaCharset.forName("UTF-8")
+  private val utf8CharsetDecoder = utf8Charset.newDecoder
+  private def decode(bytes: Array[Byte], decoder: CharsetDecoder): Try[String] = {
+    val byteByffer = ByteBuffer.wrap(bytes)
+    Try(decoder.decode(byteByffer).toString)
+  }
+  private def decode(bytes: Array[Byte], charset: JavaCharset): String = 
+    new String(bytes, charset)
+
   def apply(token: String): BasicCredentials = {
     val bytes = Base64.getDecoder.decode(token)
-    val userPass = new String(bytes, StandardCharsets.ISO_8859_1)
+    val userPass: String = decode(bytes, utf8CharsetDecoder)
+      .fold({ _ => decode(bytes, StandardCharsets.ISO_8859_1) }, identity)
```

Comment:
This looks like a suggested behavior specified in Appendix B.  It would be nice to make this whole thing configurable, but I don't see a clear path to do that yet.

Reply:
Appendix B suggests that you fallback even when there is a user-pass mismatch
```
When processing as UTF-8 fails (due to a failure to decode as UTF-8 or a mismatch of user-id/password), a server might try a fallback...
```
But IMO -  if the decoding is successful, we don't need to fallback in case of user-pass mismatch. We only need to fall back in case of an exception when decoding with utf-8. WDYT?

Reply:
Yeah, I think falling back in a user-pass mismatch would be too aggressive a default.  Furthermore, we're just decoding.  Auth is the caller's problem.

What concerns me is that if an http4s server was interpreting them as Latin1, and a client doesn't respect the new challenge parameter, we're quietly changing behavior and don't have a good recourse.  That's why I'd like to make the challenge and this behavior configurable, even though I like your new defaults.  I just don't see a great place to thread that configuration through.

Reply:
@rossabaker can we try passing an `options` or `charset` parameter to `apply` of `BasicAuth`? Since this `apply` is what returns `AuthMiddleWare`, it might be the point to start threading the configuration from.

Reply:
@rossabaker I tried passing a charset to basic auth. But like you said I am not able to wire it up all the way. Will have to spend more time on that. Do you want that implementation in this PR? Or can we do that as a separate issue?

Reply:
So sorry, I fell behind again and hadn't seen the updates here.  I think we can track that as a separate issue.  As it stands, this is a 1.0 change, but I think it's a good one.